frontend/src/routes/Map.jsx

```diff
@@ -21,6 +28,13 @@ const Map = () => {
 
   });
   const [markers, setMarkers] = useState([]);
+  const [openModal, setOpenModal] = useState('');
+  const closeModal = (e) => {
+    setOpenModal('');
+    
+    // This right here took me an hour to figure out.
+    e.stopPropagation();
```

Comment:
dang.  What was the issue?  I remember propagation has something to do with..event bubbling?

Reply:
without this, when you click close button it will call setopenmodal('') but it gets put in a queue and then the event handler bubbles up to the onclick handler that sets open modal back to the same id so it cycles lol